src/utp_socket_manager.cpp

```diff
@@ -269,21 +268,19 @@ namespace libtorrent
 	{
 		// flush all deferred acks
 
-		std::vector<utp_socket_impl*> deferred_acks;
+		static socket_vector_t deferred_acks;
+		deferred_acks.clear();
 		m_deferred_acks.swap(deferred_acks);
-		for (std::vector<utp_socket_impl*>::iterator i = deferred_acks.begin()
-			, end(deferred_acks.end()); i != end; ++i)
+		for (auto const &s : deferred_acks)
 		{
-			utp_socket_impl* s = *i;
 			utp_send_ack(s);
 		}
 
-		std::vector<utp_socket_impl*> drained_event;
+		static socket_vector_t drained_event;
```

Comment:
is the reason for having 3 different static temporary vectors that they would get mixed up with idle sockets and not provide enough pre-allocated space?

also, I imagine there should be conditions to skip this logic if ``m_deferred_acks`` and ``m_drained_events`` are empty. we don't want to swap in a pre-allocated vector from the static instance into an inactive socket.